snuba/clusters/cluster.py

```diff
@@ -66,22 +96,39 @@ def __init__(self, host: str, port: int, http_port: int, storage_sets: Set[str])
         self.__host = host
         self.__port = port
         self.__http_port = http_port
-        self.__clickhouse_rw = ClickhousePool(host, port)
-        self.__clickhouse_ro = ClickhousePool(
-            host, port, client_settings={"readonly": True},
-        )
-        self.__reader = NativeDriverReader(self.__clickhouse_ro)
+        self.__reader: Optional[Reader[SqlQuery]] = None
+        self.__connection_cache: MutableMapping[
+            Tuple[ClickhouseClientSettings, Optional[int]], ClickhousePool
+        ] = {}
 
     def __str__(self) -> str:
         return f"{self.__host}:{self.__port}"
 
-    def get_clickhouse_rw(self) -> ClickhousePool:
-        return self.__clickhouse_rw
-
-    def get_clickhouse_ro(self) -> ClickhousePool:
-        return self.__clickhouse_ro
+    def get_connection(
+        self,
+        client_settings: ClickhouseClientSettings,
+        send_receive_timeout: Optional[int] = None,
```

Comment:
I don't see this ever being used. Is this intentional ?
Also, aside from the fact that we pass this parameter during the initialization of ClickhousePool separately from the settings, is there any reason not to keep this hardcoded with the rest of the settings in ClickhouseClientSettings?
The fact that clickhouse takes settings and timeout as separate parameters is a clickhouse driver implmentation detail that should not drive our api. If we make the callsite pick the use case without specifying parameters, we should not allow the callsite to specify the timeout either.

Reply:
Hmm, yes. It's supposed to be an argument passed by user from the optimize CLI script (which is why it can't be provided on initialization). Fixing.

Reply:
Do we actually know the timeout is actually something we need? If we could remove it we will cut away quite a bit of complexity. I feel having this ad hoc parameter that is configured on the command line while all other connection parameter come from the cluster configuration is going against our design.
Could you please verify whether we absolutely need to maintain this field ?